### PR TITLE
Adding support for Windows

### PR DESCRIPTION
**Pull request by jdumouchelle**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/bld.bat [hunk 1]
```diff
@@ -0,0 +1,40 @@
+"%PYTHON%" setup.py install
+if errorlevel 1 exit 1
+
+set -o errexit
+if errorlevel 1 exit 1
+set -o pipefail
+if errorlevel 1 exit 1
+set -o nounset
+if errorlevel 1 exit 1
+
```
Comment (AntoinePrv) [inline recipe/bld.bat]: Not needed

File: recipe/bld.bat [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+cmake -B bliss-build -S "%SRC_DIR%\bliss" -D CMAKE_BUILD_TYPE=Release -D BUILD_SHARED_LIBS=OFF
```
Comment (AntoinePrv) [inline recipe/bld.bat]: Use NMake

File: recipe/bld.bat [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+if errorlevel 1 exit 1
+cmake --build bliss-build --parallel %CPU_COUNT%
+if errorlevel 1 exit 1
+cmake --install bliss-build --prefix "%PWD%\bliss-install"
+if errorlevel 1 exit 1
+
+cmake -B scipoptsuite-build -S "%SRC_DIR%\scipoptsuite" ^
```
Comment (AntoinePrv) [inline recipe/bld.bat]: Use NMake

**Review by AntoinePrv — CHANGES_REQUESTED** [review]

**Comment by AntoinePrv** [comment]
@fschloesser Did you ever encounter this error on windows before (vs2017 & boost 1.74.0)?

```error
[ 40%] Linking CXX shared library ..\..\bin\libscip.dll
LINK: command "C:\PROGRA~2\MICROS~1\2017\ENTERP~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x64\link.exe /nologo @CMakeFiles\libscip.dir\objects1.rsp /out:..\..\bin\libscip.dll /implib:..\..\lib\libscip.lib /pdb:%SRC_DIR%\scipoptsuite-build\bin\libscip.pdb /dll /version:7.0 /machine:x64 /INCREMENTAL:NO %PREFIX%\Library\lib\z.lib ..\..\lib\soplex-pic.lib %PREFIX%\Library\lib\tbb.lib %PREFIX%\Library\lib\z.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:..\..\bin\libscip.dll.manifest" failed (exit code 1120) with the following output:
   Creating library ..\..\lib\libscip.lib and object ..\..\lib\libscip.exp
presol_milp.cpp.obj : error LNK2019: unresolved external symbol "public: __cdecl boost::archive::codecvt_null<wchar_t>::codecvt_null<wchar_t>(unsigned __int64)" (??0?$codecvt_null@_W@archive@boost@@QEAA@_K@Z) referenced in function "public: void __cdecl boost::archive::codecvt_null<wchar_t>::`default constructor closure'(void)" (??_F?$codecvt_null@_W@archive@boost@@QEAAXXZ)
..\..\bin\libscip.dll : fatal error LNK1120: 1 unresolved externals
```

**Comment by fschloesser** [comment]
I've not seen it, but it could be the boost version, we test internally with boost 1.72

**Comment by AntoinePrv** [comment]
Right, I managed to make some progress with boost 1.72 and I can build SCIP.

Moving to GCG, I now get `tr1/unordered_map': No such file or directory` from `gcg/src/class_seeedpool.h`
```cpp
#if __cplusplus >= 201103L
#include <unordered_map>
using std::unordered_map;
#else
#include <tr1/unordered_map>
using std::tr1::unordered_map;
#endif
```
So I tried to compile with `CMAKE_CXX_STANDARD=14` and [`/Zc:__cplusplus`](https://docs.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=msvc-160) (since apparently it is defined to 1997), but now then I get an error in Papilo/TBB.

Not sure whether to try harder with Papilo, or just patch `gcg/src/class_seeedpool.h` to remove the check.
Do you compile with specific options on Windows internally?

**Comment by fschloesser** [comment]
@AntoinePrv We are not currently building with gcg on windows, I would propose that we turn it off in this build for now. 
I will notify the gcg devs about the problem you describe. 
What is the error in the papilo build? I added the flags and on our side it seems to compile fine.

**Comment by AntoinePrv** [comment]
Thanks @fschloesser. I deactivated windows build for gcg as I found more errors (about `#include <unistd.h>` latter on).
I don't remember the Papilo error, but I ended up leaving `__cplusplus` alone. I think everything compiles fine now, but there are still some conda related errors.

**Comment by AntoinePrv** [comment]
@jdumouchelle Why did you remove the `CPU_COUNT` ?

SCIP tests seem to be passing, which is great 🎉 

For soplex, we have
```
soplex.lib(spxdefines.cpp.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in example.cpp.obj
soplex.lib(spxdefines.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in example.cpp.obj
...
```
For which we can find information on the [Microsoft website](https://docs.microsoft.com/en-us/cpp/error-messages/tool-errors/linker-tools-error-lnk2038?view=msvc-160). I think compiling the soplex test in Release `-D CMAKE_BUILD_TYPE=Release` would help.

**Comment by jdumouchelle** [comment]
> 
> 
> @jdumouchelle Why did you remove the `CPU_COUNT` ?

On Windows it does not work.  There are some alternatives like `%NUMBER_OF_PROCESSORS%` but I still need to check if they work, so for now I just removed them.

**Comment by isuruf** [comment]
> On Windows it does not work.

It should work fine. It is set by conda-build.

**Comment by jdumouchelle** [comment]
@AntoinePrv I managed to fix also the conda related issues.  I also tested it locally and everything appears to be working fine.  I disabled zimpl as it requires gmp, which is not available for Windows on conda-forge.  

In the test scripts I had to remove the call to soplex (`soplex --version`) since no soplex.exe was created.  @fschloesser do you have any idea why the executable would not be created in Windows?   This did not effect using soplex in scip though, so I do not think this is a big issue.  

Anyway, I think the build is in a good state now and can be merged, unless anyone has any additional concerns.

**Comment by fschloesser** [comment]
@jdumouchelle I can just tell you that in our windows machine a soplex.exe is generated. I have no idea about the difference, but it could be that you are not building the standard target? However if you don't need the soplex executable it should not matter so much.

**Comment by AntoinePrv** [comment]
We build without specifying a target, so I guess that means the target is `all`.
```bat
cmake --build scipoptsuite-build --parallel "%CPU_COUNT%"
cmake --install scipoptsuite-build --prefix "%LIBRARY_PREFIX%"
```

> However if you don't need the soplex executable it should not matter so much.

It is not a priority, but given that we offer a `soplex` package, might as well have a complete one.

**Comment by fschloesser** [comment]
It looks like boost program_options is not found, which means that the soplex executable cannot be built. It also warns about this in the [windows output](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=313258&view=logs&j=a70f640f-cc53-5cd3-6cdc-236a1aa90802&t=f5d15007-a01c-5ad8-c9ce-4d519d3b275f&l=600)
```
CMake Warning at soplex/CMakeLists.txt:170 (message):
  Boost could not be found! ONLY the SoPlex library can be built without
  Boost.  The SoPlex binary will NOT be built.  If you require the binary,
  please make sure Boost is installed and your BOOST_ROOT environment
  variable is set.


```

**Comment by AntoinePrv** [comment]
Interestingly, we have afterwards
```
-- Found Boost: D:/bld/scipoptsuite_1620141112358/_h_env/Library/lib/cmake/Boost-1.72.0/BoostConfig.cmake (found version "1.72.0")  missing components: iostreams serialization program_options
```
Looking into the [`win-64/boost-cpp-1.72.0-hd4e6614_3.tar.bz2`](https://anaconda.org/conda-forge/boost-cpp/1.72.0n/download/win-64/boost-cpp-1.72.0-hd4e6614_3.tar.bz2) package, I can see the following files (and similarily for `serialization` and `iostream`)
```
Library/bin/boost_program_options.dll
Library/include/boost/program_options.hpp
Library/lib/boost_program_options.lib
Library/lib/cmake/boost_program_options-1.72.0/boost_program_options-config-version.cmake
Library/lib/cmake/boost_program_options-1.72.0/boost_program_options-config.cmake
Library/lib/cmake/boost_program_options-1.72.0/libboost_program_options-variant-shared.cmake
```
So I wonder why it is not finding it...

**Review by AntoinePrv — APPROVED** [review]

**Pull request merged**